src/lib/util/Config.ts

```diff
@@ -48,10 +49,35 @@ const DEFAULT_CONFIG: ConfigContent = {
 	]
 };
 
+class TypeChecker {
+
+	/* eslint-disable @typescript-eslint/no-explicit-any */
+	stringArrayCheck(value: any, propertyName: string, engine: ENGINE): boolean {
+		if (Array.isArray(value) && value.length > 0) {
+			value.forEach((item) => {
+				if (typeof item != 'string') {
+					throw SfdxError.create('@salesforce/sfdx-scanner', 'Config', 'OnlyStringAllowedInStringArray', [propertyName, engine.valueOf(), String(value)]);
+				}
+			});
+			return true;
+		} 
+		throw SfdxError.create('@salesforce/sfdx-scanner', 'Config', 'InvalidStringArrayValue', [propertyName, engine.valueOf(), String(value)]);
```

Comment:
Future idea. Move `@salesforce/sfdx-scanner` into Constants.ts, we are using it more often.